### PR TITLE
Create Maxcompute temporary table for training and validation

### LINKED ISSUE #568: training and validation(1): generate a temporary table for training and validation
Following design doc: training and validation
PART 1: Generate a temporary table

We add a column sqlflow_random via RAND() and save the result to a temporary table

### PR DESCRIPTION
**Pull request by weiguoz**
Fix #568

### CODE REVIEW

**Review by typhoonzero — COMMENTED** [review]
Please resolve conflicts and make CI pass

File: sql/executor.go [hunk 1]
```diff
@@ -328,6 +336,33 @@ func (cw *logChanWriter) Close() {
 	}
 }
 
+func trainWithoutValidation(tr *extendedSelect, slct string, db *DB, cwd string, wr *PipeWriter, modelDir string) error {
```
Comment (typhoonzero) [inline sql/executor.go]: This function is quite the same as `train()`, curious why we need to add this function now?

Comment (weiguoz) [reply sql/executor.go]: I filed 4 related PRs to implement the whole design. And this function used for a placeholder appears too early in this PR.
I'm taking it to the next PR.

**Review by typhoonzero — DISMISSED** [review]
LGTM++

File: sql/validation.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+var errNotSupportYet = errors.New("not support yet")
+
+// SQLFlow generates a temporary table, + sqlflow_randowm column
+func tableWithRandomColumn(db *DB, slct string) (string, error) {
+	switch db.driverName {
+	case "maxcompute":
+		return createMaxcomputeRandomTable(db, slct)
+	// TODO(weiguo): support other databases
+	case "hive", "mysql", "sqlite3":
```
Comment (Yancey0623) [inline sql/validation.go:37]: Can remove this case, we don't support other drivers except for MaxCompute.

**Review by Yancey0623 — APPROVED** [review]
LGTM with tiny comment.

**Review by typhoonzero — APPROVED** [review]
LGTM

**Pull request merged**